aom/src/main/java/com/nedap/archie/aom/primitives/CTerminologyCode.java

```diff
@@ -69,8 +69,14 @@ public boolean isValidValue(TerminologyCode value) {
                     return true;
                 }
             } else if (constraint.startsWith("ac")) {
-                if(value.getTerminologyId() != null && value.getTerminologyId().equals(constraint)) {
-                    return true;
+                // Check if the terminologyId is set in the value, and if it matches the one in the constraint.
+                // If not, get the valueSet from the archetype and check if the value is included in it's members
+                if(value.getTerminologyId() != null) {
```

Comment:
I looked at the specifications, and both terminologyId and codeString are mandatory fields, see https://www.openehr.org/releases/BASE/latest/docs/base_types/base_types.html#_terminology_code_class . So I think we should just always require the terminology id to be set. However, it's nowhere specified with what this terminology id should be filled. I would guess the ac code if present :)

Reply:
https://www.openehr.org/releases/RM/latest/docs/data_types/data_types.html#_requirements_2 also mentions the technology id "local", but I'm not sure if it is related.

Reply:
Good find! I think it should definitely be "local" in this case then. Which makes the whole 'terminology id should be an ac-code' assumption invalid anyway. @steijgeler is that something we can manage at relatively short term? :)